src/main/java/org/akaza/openclinica/controller/openrosa/QueryServiceImpl.java

```diff
@@ -467,4 +468,20 @@ public String generateDisplayId(Boolean parentDn){
             throw new OpenClinicaSystemException("Failed", ErrorConstants.ERR_GENERATING_DISCREPANCY_NOTE_ID);
         return newDisplayId;
     }
+
+    public String generateRandomNoForDisplayId() {
+        int discrepancyNoteLength;
+        try {
+             discrepancyNoteLength = Integer.parseInt(CoreResources.getField("dicrepancyNote.DisplayId.GenerationLength"));
+            if (discrepancyNoteLength <= 0 || discrepancyNoteLength > 9) {
```

Comment:
store the default value 9 in a private static final variable since you are referencing it multiple times.